src/prefect/blocks/core.py

```diff
@@ -788,9 +802,10 @@ async def _get_block_document(
         else:
             block_type_slug = cls.get_block_type_slug()
             block_document_name = name
-
         try:
-            block_document = await client.read_block_document_by_name(
+            block_document = await cast(
```

Comment:
🤦 I forgot that you can't use type annotations to change the type of a keyword argument with decorators. The current type annotations is the best you can hope for.

I'd still use `if TYPE_CHECKING: assert client is not None` instead of `cast()` because the `TYPE_CHECKING` guard is zero-cost whereas `cast()` still takes bytecode cycles for the name lookup and call.